### PR TITLE
feat: allow custom filetype support via plugins

### PR DESCRIPTION
**Pull request by marstamm**
### Proposed Changes
This PR adds the client plugin point `tabs` that allows users to create custom file type editors. This can be used to ship editors, e.g. for `.robot` files as a plugin.

The changes are fairly straight forward:

- allow expansion of `TabsProvider`
- Export commonly used Tab Components in `Plugins.js` so they can be used in Plugins
- Add Slot for additional file types on Welcome page

To allow tabs to be created, I added a new command `create-diagram` in `App.js`. This allows files to trigger `createDiagram` without hardcoding the file type in the command, which is not possible from Plugins.

You can check an example integration of the plugin point here: https://github.com/camunda/camunda-modeler/blob/RPA-spike/resources/plugins/test-tab/client/index.js

related to https://github.com/camunda/product-hub/issues/2412

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [ ] __Visual demo__ attached
* [x] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: client/src/remote/__tests__/PluginsSpec.js [hunk 1]
```diff
@@ -198,6 +198,31 @@ describe('plugins', function() {
...
+      // given
+      const global = {};
+
+      const plugins = new Plugins();
+
+      // when
+      plugins.bindHelpers(global);
+
+      // then
+      expect(global.components).to.exist;
```
[the remark this answers is not in the document]
Comment (nikku) [reply client/src/remote/__tests__/PluginsSpec.js:212]: Looks like something is missing here.

**Review by nikku — CHANGES_REQUESTED** [review]

File: client/src/remote/__tests__/PluginsSpec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (marstamm) [reply client/src/remote/__tests__/PluginsSpec.js:212]: This test simply ensures we do not accidentally remove exposed Components

[the remark this answers is not in the document]
Comment (nikku) [reply client/src/remote/__tests__/PluginsSpec.js:212]: Ok, these already exist, but exposure has not been tested yet.

[the remark this answers is not in the document]
Comment (marstamm) [reply client/src/remote/__tests__/PluginsSpec.js:212]: Correct, the exposed components were untested before

[the remark this answers is not in the document]
Comment (nikku) [reply client/src/remote/__tests__/PluginsSpec.js:212]: (Added in previous commit, I propose to move the test over to https://github.com/camunda/camunda-modeler/pull/4568/commits/a8e590058f370428545bc62ce49e0dafdbd87f74).

[the remark this answers is not in the document]
Comment (marstamm) [reply client/src/remote/__tests__/PluginsSpec.js:212]: Oh, seems I squashed the wrong commits 😅 

will adjust the commit history accordingly

**Review by nikku — APPROVED** [review]

**Pull request merged**